blog/_posts/2021-06-17-uncertainty.md

```diff
@@ -0,0 +1,244 @@
+the predicted class and therefore can increase calibration as shown in the following figure.
+
+![](/blog/images/uncertainty/calibration3.jpg)
+
+## Ensembles
+
+Ensemble of neural networks can be a straightforward way of estimating uncertainty by quantifying the disagreement 
+between each model in the ensemble. Lakshminarayanan et al. propose adding adversarial examples to this training 
+procedure [^8]. 
+
+![](/blog/images/uncertainty/advensemble.jpg)
+
+
+## Bayesian Neural Networks
+
+#### Bayes by Backprob
+
+Blundell et al. introduce a way of training Bayesian neural networks known as *Bayes by Backprop* in 
+Weight Uncertainty in Neural Networks [^1]. Instead of having a point estimate for each weight, Bayesian neural 
+networks learn a distribution over each parameter.
+
+![](/blog/images/uncertainty/bayesbybackprop.jpg)
+
+The goal of Bayesian neural networks is to find the posterior distribution of the weights, which can be expressed with 
+Baye's theorem: 
+
+$$
+P(w|D) = \frac{P(D|w)P(w)}{P(D)}
+$$
+
+Given this distribution and a new sample $$x^*$$, we can get a predictive distribution for $$y^*$$: 
+
+$$
+P(y^*|x^*, D) = \int P(y^* | x^*, w) P(w|D) dw
+$$
+
+However, there is no tractable analytical solution for $$P(w|D)$$ for neural networks. Variational inference is therefore
+used to estimate the true posterior with a variational approximation $$q(w|\theta)$$. This gives the following loss 
+function:
+
+$$
+\theta^* = \argmin_\theta KL[ q(w|\theta) || P(w|D) ] 
+$$
+
+$$
+\theta^* = \argmin_\theta \int  q(w|\theta) \log \frac{q(w|\theta)}{P(w)P(w|D)} 
+$$
+
+$$
+\theta^* = \argmin_\theta  KL[ q(w|\theta) || P(w) ] - \mathbb{E}_q(w|\theta)[\log P(w|D)] 
+$$
+
+
+#### MC Dropout
+Training Bayesian neural networks is not trivial and requires substantial changes to the training procedure. Gal et al. 
+show that neural networks with dropout can be used as an approximation for Bayesian nets [^2]. By using dropout at test time, one can
+generate a Monte Carlo distribution of predictions which can be used to estimate the uncertainty of the prediction.   
+
+#### What uncertainties are needed?
+Kendall et al. describe two types of uncertainty in *What Uncertainties Do We Need in Bayesian Deep
+Learning for Computer Vision?* [^7]. Firstly, epistemic uncertainty is the uncertainty in the model weights. This can be
+represented with Bayesian nets or MC dropout. This type of uncertainty can be decreased with more data. The second type 
+of uncertainty is aleatoric uncertainty; the uncertainty in the data. This type of uncertainty cannot be 
+reduced by more data as it is inherent to the data. Aleatoric uncertainty can be homoscedastic (constant for all 
+inputs) or heteroscedastic (dependant on the input sample). 
+
+As mentioned estimating epistemic uncertainty can be done using MC dropout. The prediction is given by: 
```

Comment:
`As mentioned estimating epistemic uncertainty can be done (...)`: `As mentioned, decreasing epistemic uncertainty can be done (...)` ?

Reply:
estimating epistemic uncertainty is done with MC dropout, to reduce it you must add more data.

Reply:
OK.